--- ../src-base/minecraft/net/minecraft/item/ItemWritableBook.java
+++ ../src-work/minecraft/net/minecraft/item/ItemWritableBook.java
@@ -3,6 +3,7 @@
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.nbt.NBTTagList;
+import net.minecraft.nbt.NBTTagString;
 import net.minecraft.world.World;
 
 public class ItemWritableBook extends Item
@@ -25,6 +26,22 @@
         return true;
     }
 
+    // Rus-Minecraft: compound update
+    public static boolean shadowUpdate(ItemStack stack, ItemStack push, boolean isSigning) {
+        NBTTagCompound pushNBT = push.hasTagCompound() ? push.getTagCompound() : new NBTTagCompound();
+        NBTTagCompound stackNBT = stack.hasTagCompound() ? stack.getTagCompound() : new NBTTagCompound();
+        if (isSigning) {
+            if (pushNBT.hasKey("title"))
+                stackNBT.setTag("title", pushNBT.getTag("title"));
+            if (pushNBT.hasKey("author"))
+                stackNBT.setTag("author", pushNBT.getTag("author"));
+        }
+        if (pushNBT.hasKey("pages"))
+            stackNBT.setTag("pages", pushNBT.getTag("pages"));
+        stack.setTagCompound(stackNBT);
+        return true;
+    }
+
     public static boolean func_150930_a(NBTTagCompound p_150930_0_)
     {
         if (p_150930_0_ == null)
@@ -43,15 +60,10 @@
             {
                 String s = nbttaglist.getStringTagAt(i);
 
-                if (s == null)
+                if (s == null || s.length() > 256)
                 {
                     return false;
                 }
-
-                if (s.length() > 256)
-                {
-                    return false;
-                }
             }
 
             return true;
